packages/core/src/components/Sequence/Step.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { FC } from 'react';
+import {
+  Step as MuiStep,
+  StepContent,
+  StepLabel,
+  Typography,
+  makeStyles,
+} from '@material-ui/core';
+import StepperFooter from './StepperFooter';
+
+const useStyles = makeStyles(theme => ({
+  end: {
+    padding: theme.spacing(3),
+  },
+}));
+
+export type StepActions = {
+  showNext?: boolean;
+  canNext?: () => boolean;
+  onNext?: () => void;
+  nextStep?: (current: number, last: number) => number;
+  nextText?: string;
+
+  showBack?: boolean;
+  backText?: string;
+  onBack?: () => void;
+
+  showRestart?: boolean;
+  canRestart?: () => boolean;
+  onRestart?: () => void;
+  restartText?: string;
+};
+
+type StepProps = {
+  title: string;
+  children: React.ReactElement;
+  stepperLength?: number;
+  stepIndex?: number;
+  setStepIndex?: any;
+  stepArray?: number[];
+  setStepArray?: any;
+  end?: boolean;
+  onStepperStepChange?: (prevIndex: number, nextIndex: number) => void;
+  actions?: StepActions;
+};
+
+const Step: FC<StepProps> = ({
+  title,
+  children,
+  end,
+  stepperLength,
+  stepIndex,
+  setStepIndex,
```

Comment:
The three items above may be seen as power features perhaps? I don't mind much tho

Reply:
Oh now I think I see - these are part of the public API of the Step even though the user is never supposed to set them, is that right?

Could you wrap the Stepper in a react context instead, that manages the state and actions handling what the current step is etc? The user should only be exposed to the exact props that they can affect.

Reply:
yea, they're not exposed to this. The prop is managed in `Stepper` and cloned into its children (`Step`)

Reply:
i'll break apart the prop typing to make this clear, cause you're right its not clear at all

Reply:
also document it. but the user facing props are title, children and actions

Reply:
ah yeap context makes much more sense @freben, my brain wasn't working